### PR TITLE
[Bug 18856] Use string number cache for number formatting/parsing

### PR DESCRIPTION
**Pull request by peter-b**

### CODE REVIEW

**Comment by livecodefraser** [comment]
@livecode-vulcan review okay 6e6ff8e

File: libscript/src/module-arithmetic.cpp [hunk 1]
```diff
@@ -473,9 +475,11 @@ extern "C" MC_DLLEXPORT_DEF MCStringRef MCArithmeticExecFormatNumberAsString(MCN
 extern "C" MC_DLLEXPORT_DEF MCValueRef MCArithmeticExecParseStringAsNumber(MCStringRef p_operand)
 {
     double t_converted;
-    if (!MCTypeConvertStringToReal(p_operand, t_converted))
+
+    if (!(MCStringGetNumericValue(p_operand, t_converted) ||
+          MCTypeConvertStringToReal(p_operand, t_converted)))
```
Comment (livecodeali) [inline libscript/src/module-arithmetic.cpp]: Doesn't formulating it like this mean `MCTypeConvertStringToReal` is called even if `MCStringGetNumericValue` fails?

**Comment by livecode-vulcan** [comment]
💙 review by @livecodefraser ok 6e6ff8ed2abec6cff979f8fe0890560c47715dc2

**Comment by livecode-vulcan** [comment]
😎 test success 6e6ff8ed2abec6cff979f8fe0890560c47715dc2
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/607)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/612)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/607)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/611)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/616)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/608)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/608)

**Pull request merged**